cdap-examples/WikipediaPipeline/src/main/scala/co/cask/cdap/examples/wikipedia/ClusteringUtils.scala

```diff
@@ -98,15 +98,15 @@ object ClusteringUtils {
    * Store the clustering results in the specified dataset and update accumulators and workflow token.
    */
   def storeResults(sc: SparkContext, sec: SparkExecutionContext,
-                   results: Array[Array[(String, Double)]], tableName: String) = {
+                   results: Array[Array[(String, Double)]], namespace: String, tableName: String) = {
     val numRecords: Accumulator[Int] = sc.accumulator(0, "num.records")
     val initial= new Term("", 0.0)
     val highestScore: Accumulator[Term] = sc.accumulator(initial, "highest.score")(HighestAccumulatorParam)
 
     sec.execute(new TxRunnable {
       override def run(context: DatasetContext) = {
-        val table: Table = context.getDataset(tableName)
-
+        var table: Table = if (namespace != null) context.getDataset(namespace, tableName)
```

Comment:
Yikes!! this kind of if else looks ugly. But if there is no ternary then its fine.

I saw your comment earlier comment about not being able to find getNamespace API in Service which I see you found now.
Look into docs. Will this not work ? 

```
    val dataNamespace = if (sec.getRuntimeArguments.get(WikipediaPipelineApp.NAMESPACE_ARG) == null) sec.getNamespace 
    else sec.getRuntimeArguments.get(WikipediaPipelineApp.NAMESPACE_ARG) 
```